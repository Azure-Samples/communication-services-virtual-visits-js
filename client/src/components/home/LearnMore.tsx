// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { Icon, Link, Stack, StackItem, Text } from '@fluentui/react';
import {
  font12pxSemiBoldStyle,
  font12pxWeight400Style,
  font16pxStyle,
  newWindowIconWrapper,
  textDecorationNone
} from '../../styles/Home.styles';

const LearnMore = (): JSX.Element => {
  return (
    <>
      <Text styles={font16pxStyle}>Learn more about Azure Communication Services</Text>
      <LearnMoreItem
        headerText={'Azure Communication Services virtual appointments'}
        headerLink={'https://learn.microsoft.com/azure/communication-services/tutorials/virtual-visits'}
        description={
          'This tutorial describes concepts for virtual appointment applications. After completing this tutorial and the associated Sample Builder, you will understand common use cases that a virtual appointment delivers...'
        }
      />
      <LearnMoreItem
        headerText={'Azure Communication Services Rooms (Preview)'}
        headerLink={'https://learn.microsoft.com/en-us/azure/communication-services/concepts/rooms/room-concept'}
        description={
          'Azure Communication Services provides a concept of a room for developers who are building structured conversations such as virtual appointments or virtual events. Rooms currently allow voice and video calling...'
        }
      />
      <LearnMoreItem
        headerText={'Get Started with UI Library'}
        headerLink={
          'https://learn.microsoft.com/azure/communication-services/quickstarts/ui-library/get-started-composites?tabs=kotlin&pivots=platform-web'
        }
        description={
          'Get Started with Azure Communication Services UI Library to quickly integrate communication experiences into your applications. In this quickstart, learn how to integrate UI Library composites into an application and set up...'
        }
      />
    </>
  );
};

interface LearnMoreItemProps {
  headerText: string;
  headerLink: string;
  description: string;
}

const LearnMoreItem: React.FunctionComponent<LearnMoreItemProps> = (props: LearnMoreItemProps) => {
  return (
    <>
      <Link target="_blank" href={props.headerLink}>
        <Stack horizontal>
          <StackItem align="center" style={font12pxSemiBoldStyle}>
            {props.headerText}
          </StackItem>
          <StackItem align="center" style={textDecorationNone}>
            <span style={newWindowIconWrapper}>
              <Icon iconName="OpenInNewWindow" />
            </span>
          </StackItem>
        </Stack>
      </Link>
      <Text styles={font12pxWeight400Style}>{props.description}</Text>
    </>
  );
};

export default LearnMore;
