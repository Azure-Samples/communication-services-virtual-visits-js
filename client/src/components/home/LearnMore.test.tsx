// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import renderer from 'react-test-renderer';
import LearnMore from './LearnMore';

describe('LearnMore', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<LearnMore />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
