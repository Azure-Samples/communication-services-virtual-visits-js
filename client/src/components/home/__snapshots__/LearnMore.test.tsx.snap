// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnMore should match snapshot 1`] = `
Array [
  <span
    className="css-109"
  >
    Learn more about Azure Communication Services
  </span>,
  <a
    className="ms-Link root-110"
    href="https://learn.microsoft.com/azure/communication-services/tutorials/virtual-visits"
    onClick={[Function]}
    target="_blank"
  >
    <div
      className="ms-Stack css-111"
    >
      <div
        className="ms-StackItem css-112"
        style={
          Object {
            "fontSize": ".75rem",
            "fontWeight": "600",
            "lineHeight": "1rem",
            "paddingRight": ".5rem",
            "verticalAlign": "bottom",
          }
        }
      >
        Azure Communication Services virtual appointments
      </div>
      <div
        className="ms-StackItem css-112"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <span
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <i
            aria-hidden={true}
            className="root-105"
            data-icon-name="OpenInNewWindow"
          />
        </span>
      </div>
    </div>
  </a>,
  <span
    className="css-113"
  >
    This tutorial describes concepts for virtual appointment applications. After completing this tutorial and the associated Sample Builder, you will understand common use cases that a virtual appointment delivers...
  </span>,
  <a
    className="ms-Link root-110"
    href="https://learn.microsoft.com/en-us/azure/communication-services/concepts/rooms/room-concept"
    onClick={[Function]}
    target="_blank"
  >
    <div
      className="ms-Stack css-111"
    >
      <div
        className="ms-StackItem css-112"
        style={
          Object {
            "fontSize": ".75rem",
            "fontWeight": "600",
            "lineHeight": "1rem",
            "paddingRight": ".5rem",
            "verticalAlign": "bottom",
          }
        }
      >
        Azure Communication Services Rooms (Preview)
      </div>
      <div
        className="ms-StackItem css-112"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <span
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <i
            aria-hidden={true}
            className="root-105"
            data-icon-name="OpenInNewWindow"
          />
        </span>
      </div>
    </div>
  </a>,
  <span
    className="css-113"
  >
    Azure Communication Services provides a concept of a room for developers who are building structured conversations such as virtual appointments or virtual events. Rooms currently allow voice and video calling...
  </span>,
  <a
    className="ms-Link root-110"
    href="https://learn.microsoft.com/azure/communication-services/quickstarts/ui-library/get-started-composites?tabs=kotlin&pivots=platform-web"
    onClick={[Function]}
    target="_blank"
  >
    <div
      className="ms-Stack css-111"
    >
      <div
        className="ms-StackItem css-112"
        style={
          Object {
            "fontSize": ".75rem",
            "fontWeight": "600",
            "lineHeight": "1rem",
            "paddingRight": ".5rem",
            "verticalAlign": "bottom",
          }
        }
      >
        Get Started with UI Library
      </div>
      <div
        className="ms-StackItem css-112"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <span
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <i
            aria-hidden={true}
            className="root-105"
            data-icon-name="OpenInNewWindow"
          />
        </span>
      </div>
    </div>
  </a>,
  <span
    className="css-113"
  >
    Get Started with Azure Communication Services UI Library to quickly integrate communication experiences into your applications. In this quickstart, learn how to integrate UI Library composites into an application and set up...
  </span>,
]
`;
